go/proof.go

```diff
@@ -217,6 +217,10 @@ func IsLeftMost(spec *InnerSpec, path []*InnerOp) bool {
 	// ensure every step has a prefix and suffix defined to be leftmost
 	for _, step := range path {
 		if !hasPadding(step, minPrefix, maxPrefix, suffix) {
+			// if this is a placeholder node, skip it
+			if isPlaceholder(spec, 0, step, true) {
```

Comment:
These calls make sense.

Since most of the logic is inside an if statement, why not make two functions - `areLeftBranchesEmpty` and `areRightBranchesEmpty` or something like that?

Reply:
when I started writing the function, I thought the paths might share some code - but yes, I can just split it up.